src/main/ui/src/components/Login/Login.jsx

```diff
@@ -0,0 +1,89 @@
+import React, { useState, useEffect } from "react";
+import basestyle from "./Base.module.css";
+import loginstyle from "./Login.module.css";
+import axios from "axios";
+import { useNavigate, NavLink } from "react-router-dom";
+const Login = ({ setUserState }) => {
+  const navigate = useNavigate();
+  const [formErrors, setFormErrors] = useState({});
+  const [isSubmit, setIsSubmit] = useState(false);
+  const [user, setUserDetails] = useState({
+    email: "",
+    password: "",
+  });
+
+  const changeHandler = (e) => {
+    const { name, value } = e.target;
+    setUserDetails({
+      ...user,
+      [name]: value,
+    });
+  };
+  const validateForm = (values) => {
+    const error = {};
+    const regex = /^[^\s+@]+@[^\s@]+\.[^\s@]{2,}$/i;
+      if (!values.fname) {
+          error.fname = "First Name is required";
+        }
+    if (!values.email) {
+      error.email = "Email is required";
+    } else if (!regex.test(values.email)) {
+      error.email = "Please enter a valid email address";
+    }
+    if (!values.password) {
+      error.password = "Password is required";
+    }
+    return error;
+  };
+
+  const loginHandler = (e) => {
+    e.preventDefault();
+    setFormErrors(validateForm(user));
+    setIsSubmit(true);
+    // if (!formErrors) {
+
+    // }
+  };
+
+  useEffect(() => {
+    if (Object.keys(formErrors).length === 0 && isSubmit) {
+      console.log(user);
+      axios.post("http://localhost:3000/login", user).then((res) => {
+        alert(res.data.message);
```

Comment:
No alerts. It interrupts the user experience and stops all functions on a SPA app. can log errors to the console or if it's being relayed to the user by a css error class. Bootstrap should be able to help if it's an error the user should see.